flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV110.java

```diff
@@ -294,4 +296,19 @@ public boolean isDateStats(ColumnStatisticsData colStatsData) {
 	public CatalogColumnStatisticsDataDate toFlinkDateColStats(ColumnStatisticsData hiveDateColStats) {
 		throw new UnsupportedOperationException("DATE column stats are not supported until Hive 1.2.0");
 	}
+
+	@Override
+	public Set<String> listBuiltInFunctions() {
+		// FunctionInfo doesn't have isBuiltIn() API to tell whether it's a builtin function or not
+		// prior to Hive 1.2.0
+		throw new UnsupportedOperationException("Listing built in functions are not supported until Hive 1.2.0");
```

Comment:
Can we check `FunctionInfo.isNative` to decide whether a function is built-in or not? It seems user defined functions are registered using `FunctionRegistry.registerTemporaryFunction` in [FunctionTask](https://github.com/apache/hive/blob/release-1.1.1/ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionTask.java#L72), in which case it's considered non-native.

Reply:
no, persistent function is also native